POST to content-property without content-type results in NPE / HTTP-500

**Describe the bug**

(this is a minor low-priority bug)

User/developer uses Spring Content REST and sends a POST to a content property, but forgets to set Content-Type header.

This results in an HTTP-500 with this stacktrace:
```
Request processing failed; nested exception is java.lang.NullPointerException: Cannot invoke "org.springframework.http.MediaType.toString()" because "sourceMimeType" is null
org.springframework.web.util.NestedServletException: Request processing failed; nested exception is java.lang.NullPointerException: Cannot invoke "org.springframework.http.MediaType.toString()" because "sourceMimeType" is null
[...snip...]
Caused by: java.lang.NullPointerException: Cannot invoke "org.springframework.http.MediaType.toString()" because "sourceMimeType" is null
	at internal.org.springframework.content.rest.contentservice.ContentStoreContentService.setContent(ContentStoreContentService.java:155)
	at internal.org.springframework.content.rest.controllers.StoreRestController.handleMultipart(StoreRestController.java:217)
[...snip...]
```

**Expected behavior**
HTTP-400

**Steps to reproduce**

```
@Test
void postInvoiceContent_missingContentType_http400() throws Exception {
    mockMvc.perform(post("/invoices/{id}/content", invoiceNumber).content("some content"))
        .andExpect(status().isBadRequest());
}
```

--- a/spring-content-rest/src/main/java/internal/org/springframework/content/rest/contentservice/ContentStoreContentService.java
+++ b/spring-content-rest/src/main/java/internal/org/springframework/content/rest/contentservice/ContentStoreContentService.java
@@ -1,19 +1,14 @@
 package internal.org.springframework.content.rest.contentservice;
 
-import static java.lang.String.format;
-
-import java.io.File;
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.Serializable;
-import java.lang.reflect.Method;
-import java.nio.file.Files;
-import java.util.*;
-
+import internal.org.springframework.content.rest.controllers.BadRequestException;
+import internal.org.springframework.content.rest.controllers.MethodNotAllowedException;
+import internal.org.springframework.content.rest.io.AssociatedStoreResource;
+import internal.org.springframework.content.rest.io.RenderedResource;
+import internal.org.springframework.content.rest.io.StoreResource;
+import internal.org.springframework.content.rest.mappingcontext.ContentPropertyToExportedContext;
+import internal.org.springframework.content.rest.mappings.StoreByteRangeHttpRequestHandler;
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
-
-import internal.org.springframework.content.rest.mappingcontext.ContentPropertyToExportedContext;
 import org.apache.commons.io.FileUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
@@ -42,11 +37,15 @@ import org.springframework.util.ReflectionUtils;
 import org.springframework.util.StringUtils;
 import org.springframework.web.server.ResponseStatusException;
 
-import internal.org.springframework.content.rest.controllers.MethodNotAllowedException;
-import internal.org.springframework.content.rest.io.AssociatedStoreResource;
-import internal.org.springframework.content.rest.io.RenderedResource;
-import internal.org.springframework.content.rest.io.StoreResource;
-import internal.org.springframework.content.rest.mappings.StoreByteRangeHttpRequestHandler;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.util.*;
+
+import static java.lang.String.format;
 
 public class ContentStoreContentService implements ContentService {
 
@@ -146,6 +145,10 @@ public class ContentStoreContentService implements ContentService {
 
     @Override
     public void setContent(HttpServletRequest request, HttpServletResponse response, HttpHeaders headers, Resource source, MediaType sourceMimeType, Resource target) throws IOException, MethodNotAllowedException {
+
+        if (sourceMimeType == null) {
+            throw new BadRequestException("Missing Content-Type header");
+        }
 
         AssociatedStoreResource storeResource = (AssociatedStoreResource)target;
         ContentProperty property = storeResource.getContentProperty();

--- a/spring-content-rest/src/test/java/it/internal/org/springframework/content/rest/controllers/Content.java
+++ b/spring-content-rest/src/test/java/it/internal/org/springframework/content/rest/controllers/Content.java
@@ -298,6 +298,15 @@ public class Content {
                     assertThat(fetched.get().getLen(), is(Long.valueOf(content.length())));
                 });
             });
+            Context("a POST to /{store}/{id} with a missing content-type", () -> {
+                It("should return 400 Bad Request", () -> {
+                    mvc.perform(post(url)
+                                    .content("some content")
+                                    .contextPath(contextPath)
+                            )
+                            .andExpect(status().isBadRequest());
+                });
+            });
             Context("a PUT to /{store}/{id} with a multi-part request", () -> {
                 It("should overwrite the content and return 200", () -> {
 

